platform/kube/controller.go

```diff
@@ -131,6 +134,25 @@ func NewController(client *Client, options ControllerOptions) *Controller {
 			func(opts meta_v1.ListOptions) (watch.Interface, error) {
 				return client.client.ExtensionsV1beta1().Ingresses(options.Namespace).Watch(opts)
 			})
+
+		// Ingress sync mode -> ingress status sync mode. The ingress status is synced in two cases:
+		// 1. default ingress class provided and no annotation on ingress resource
+		// 2. specific ingress class provided and same annotation on ingress resource
+		var ingressClass, defaultIngressClass string
+		switch options.IngressSyncMode {
+		case IngressStrict:
+			ingressClass = options.IngressClass
+		case IngressDefault:
+			defaultIngressClass = options.IngressClass
+		}
+
+		out.ingressStatus = status.NewStatusSyncer(status.Config{
```

Comment:
Regarding the silly `POD_IP` env variable use by the library, we should file an issue and do a hack to set the env variable for now. I think maybe add it `cmd/main.go` since there are similar flags there already.